src/framework/mlt_frame.c

```diff
@@ -445,6 +470,9 @@ int mlt_image_format_size( mlt_image_format format, int width, int height, int *
 		case mlt_image_glsl_texture:
 			if ( bpp ) *bpp = 0;
 			return 4;
+		case mlt_image_yuv422p16:
+			if ( bpp ) *bpp = 2;
+			return height * ( UP16( width * 2 ) + 2 * UP16( width + 1 ) ) ;
```

Comment:
Yes, I agree. Images and image rows on mlt_frame are not required to be 16 byte aligned. MLT framework does not impose these requirements on its services such that implementations of signal processing can be simper and less optimized. Major rework to deal with all that would be required. I think it is great that FFmpeg has support and optimizations for these things, but this is not FFmpeg, and I do not have the time and energy to recreate what they have done (or ensure someone else's changes does not break shit). Also, maybe there should be mlt_image_avformat and additional frame properties that allows any FFmpeg pixel format to be used only within the avformat module.